### PR TITLE
feat(credit): Add simple credit record to transaction log and compute balance to user profile.

### PR DESCRIPTION
**Pull request by janbarasek**
Snímek obrazovky 2024-10-04 v 12 23 02

### CODE REVIEW

File: libs/api/utils/src/errorList.ts [hunk 1]
```diff
@@ -50,4 +50,8 @@ export const ERROR_LIST = {
     code: 401,
     message: 'Organisation API key has been expired.',
   },
+  USER_CREDIT_CAN_NOT_BE_NEGATIVE: {
+    code: 500,
```
Comment (BleedingDev) [inline libs/api/utils/src/errorList.ts:54]: **thought:** Is it really our fault (error 500), when it's user's credits problem? I think 402 would be better. Or anything from 4xx family.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/402

File: libs/db/src/schema.prisma [hunk 2]
```diff
@@ -115,8 +115,11 @@ model cas__user {
   first_name                String?                     @db.VarChar(100)
   last_name                 String?                     @db.VarChar(100)
   avatar_url                String?
+  credit                    Int
+  credit_alert_lower_then   Int?
```
Comment (BleedingDev) [inline libs/db/src/schema.prisma:119]: **question:** Will it be hard to have multiple thresholds if we allow users to have multiple triggers? Now I'd say let's keep it simple, it's just a thought.

**Review by BleedingDev — APPROVED** [review]
Only one meaningful change request, but nothing mandatory, so approving.

File: libs/db/src/schema.prisma (hunk 2 above)
Comment (janbarasek) [reply libs/db/src/schema.prisma:119]: It can be refactored in future.

**Review by BleedingDev — APPROVED** [review]

**Pull request merged**